samples/snippets/append_rows_proto2.py

```diff
@@ -0,0 +1,227 @@
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    row = sample_data_pb2.SampleData()
+    row.string_col = "Auf Wiedersehen!"
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    request = types.AppendRowsRequest()
+    request.write_stream = stream_name
+    proto_data = types.AppendRowsRequest.ProtoData()
+    proto_data.rows = proto_rows
+
+    # Generate a protocol buffer representation of your message descriptor. You
+    # must inlcude this information in the first request of an append_rows
+    # stream so that BigQuery knows how to parse the serialized_rows.
+    proto_schema = types.ProtoSchema()
+    proto_descriptor = descriptor_pb2.DescriptorProto()
+    sample_data_pb2.SampleData.DESCRIPTOR.CopyToProto(proto_descriptor)
+    proto_schema.proto_descriptor = proto_descriptor
+    proto_data.writer_schema = proto_schema
+    request.proto_rows = proto_data
+
+    yield request
+
+    # Create a batch of rows containing scalar values that don't directly
+    # correspond to a protocol buffers scalar type. See the documentation for
+    # the expected data formats:
+    # https://cloud.google.com/bigquery/docs/write-api#data_type_conversions
+    proto_rows = types.ProtoRows()
+
+    row = sample_data_pb2.SampleData()
+    date_value = datetime.date(2021, 8, 12)
+    epoch_value = datetime.date(1970, 1, 1)
+    delta = date_value - epoch_value
+    row.date_col = delta.days
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    row = sample_data_pb2.SampleData()
+    datetime_value = datetime.datetime(2021, 8, 12, 9, 46, 23, 987456)
+    row.datetime_col = datetime_value.strftime("%Y-%m-%d %H:%M:%S.%f")
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    row = sample_data_pb2.SampleData()
+    row.geography_col = "POINT(-122.347222 47.651111)"
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    row = sample_data_pb2.SampleData()
+    numeric_value = decimal.Decimal("1.23456789101112e+6")
+    row.numeric_col = str(numeric_value)
+    bignumeric_value = decimal.Decimal("-1.234567891011121314151617181920e+16")
+    row.bignumeric_col = str(bignumeric_value)
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    row = sample_data_pb2.SampleData()
+    time_value = datetime.time(11, 7, 48, 123456)
+    row.time_col = time_value.strftime("%H:%M:%S.%f")
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    row = sample_data_pb2.SampleData()
+    timestamp_value = datetime.datetime(
+        2021, 8, 12, 16, 11, 22, 987654, tzinfo=datetime.timezone.utc
+    )
+    epoch_value = datetime.datetime(1970, 1, 1, tzinfo=datetime.timezone.utc)
+    delta = timestamp_value - epoch_value
+    row.timestamp_col = int(delta.total_seconds()) * 1000000 + int(delta.microseconds)
+    proto_rows.serialized_rows.append(row.SerializeToString())
+
+    # Since this is the second request, you only need to include the row data
```

Comment:
you should only need the row data, streamid/schema/traceid all get set on first request in the stream.

Reply:
You're right. Omitting the stream name on subsequent requests worked fine.